### DISCUSSION TITLE
Invalid Route Base on 0.5-rc.1

### DISCUSSION THREAD

**Discussion by HamishWHC** [rwf2/Rocket#1767] [Questions] [answered]
Trying to build a project with a dynamic path handler. Have found that I can't build even this route:
```rust
#[get("/hello/<name>")]
fn hello(name: &str) -> String {
    format!("Hello, {}!", name)
}
```
I get this error:
```
Error: invalid route base: /hello/<name>
   >> unexpected token '<' at index 7
```
What am I missing?

**Comment by HamishWHC** [comment]
Extra info: I can compile with static paths just fine. I'm on WSL 2. Can't seem to compile the hello example with this error:
```
error: /home/hamishwhc/projects/Rocket/examples/target/debug/deps/librocket_codegen-c6b35877a4b83c8c.so: undefined symbol: _ZN5alloc7raw_vec19RawVec$LT$T$C$A$GT$11allocate]in17h47bab2c373b57925E
  --> /home/hamishwhc/projects/Rocket/core/lib/src/response/mod.rs:30:9
   |
30 | pub use rocket_codegen::Responder;
   |  
```
Although if I compile the example on Windows instead of WSL, it runs just fine. I'll try my project on Windows, but would prefer to stay in WSL.

**Comment by SergioBenitez** [comment, marked as answer]
You're not showing us the rest of your code. The error indicates that you're writing `.mount("/hello/<name>", ...)` elsewhere, which is completely malformed. Even a cursory reading of the documentation, or a proper copying of any of the myriad examples, would resolve and answer your question. Please read the documentation.

**Reply by HamishWHC** [reply]
Here you go:
```rust
#[macro_use] extern crate rocket;

// pub mod api;
// pub mod models;
pub mod cors;

// use api::{
//     login::*,
//     logout::*,
//     user::*
// };

#[get("/hello/<name>")]
fn hello(name: &str) -> String {
    format!("Hello, {}!", name)
}

#[launch]
async fn rocket() -> _ {
    let database_url = "sqlite://../test.db";
    let pool = sqlx::SqlitePool::connect(database_url)
        .await
        .expect("Failed to connect to database");

    rocket::build()
        .attach(cors::fairing())
        .manage(pool)
        .mount("/", routes![
            hello,
            // login,
            // logout,
            // user_info,
            //user_profile
        ])
}
```

**Reply by SergioBenitez** [reply]
What is your `cors::fairing()` doing?

**Reply by HamishWHC** [reply]
I see. Just checked the CORS fairing. IDK how its meant to work, but it is running mount on every route...

Would it be possible to improve the error message? At least to suggest checking for .mount calls.

**Reply by SergioBenitez** [reply]
The error message should be indicating more information. Is that really all of the log output that you get? Could you post everything your terminal is showing you?

In the future, you'll want to provide us with more information than two sparse snippets of code and terminal output.

**Reply by HamishWHC** [reply]
Here's the full terminal log:
```
❯ cargo run
   Compiling tournament-api v0.1.0 (/home/hamishwhc/projects/tournament-infra/api)
    Finished dev [unoptimized + debuginfo] target(s) in 19.77s
     Running `target/debug/tournament-api`
Error: invalid route base: /hello/<name>
   >> unexpected token '<' at index 7
thread 'main' panicked at 'aborting due to route base error', /home/hamishwhc/.cargo/registry/src/github.com-1ecc6299db9ec823/rocket-0.5.0-rc.1/src/rocket.rs:227:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Thanks for your help.

**Reply by SergioBenitez** [reply]
Are you suppressing log output? What does your Rocket.toml look like?

**Reply by HamishWHC** [reply]
Rocket.toml:
```toml
[development]
address = "localhost"
port = 9989
```

**Reply by SergioBenitez** [reply]
We should definitely be providing more context for the error. I'll look into it soon.
